WordPress/Sniffs/PHP/EnqueuedCheckSniff.php

```diff
@@ -0,0 +1,132 @@
+	 *
+	 * @since 0.11.0
+	 * @var string
+	 */
+	protected $group_name = 'Enqueued';
+
+	/**
+	 * List of array functions to which a $strict parameter can be passed.
+	 *
+	 * The $strict parameter is the third and last parameter for each of these functions.
+	 *
+	 * The array_keys() function only requires the $strict parameter when the optional
+	 * second parameter $search has been set.
+	 *
+	 * @link http://php.net/in-array
+	 * @link http://php.net/array-search
+	 * @link http://php.net/array-keys
+	 *
+	 * @since 0.10.0
+	 * @since 0.11.0 Renamed from $array_functions to $target_functions.
+	 *
+	 * @var array <string function_name> => <bool always needed ?>
+	 */
+	
+	/**
+	 * List of Enqueued functions that need to be check to make sure
+	 * IF a source ($src) value is passed, then version ($ver) needs to have a value.
+	 * Additionally, IF a source ($src) value is passed a check for In footer ($in_footer) 
+	 * to alert the user if the value isnt True
+	 * 
+	 * @link https://developer.wordpress.org/reference/functions/wp_register_script/
+	 * @link https://developer.wordpress.org/reference/functions/wp_enqueue_script/
+	 * @link https://developer.wordpress.org/reference/functions/wp_register_style/
+	 * @link https://developer.wordpress.org/reference/functions/wp_enqueue_style/
+	 *
+	 * @since 0.10.0
+	 * @since 0.11.0 Renamed from $array_functions to $target_functions.
+	 *
+	 * @var array <string function_name> => <bool always needed ?>
+	 */
+	protected $target_functions = array(
+		'wp_register_script' => true,
+		'wp_enqueue_script'  => true,
+		'wp_register_style'  => true,
+		'wp_enqueue_style'   => true
+	);
+
+	/**
+	 * Process the parameters of a matched function.
+	 *
+	 * @since 0.11.0
+	 *
+	 * @param int    $stackPtr        The position of the current token in the stack.
+	 * @param array  $group_name      The name of the group which was matched.
+	 * @param string $matched_content The token content (function name) which was matched.
+	 * @param array  $parameters      Array with information about the parameters.
+	 *
+	 * @return void
+	 */
+	public function process_parameters( $stackPtr, $group_name, $matched_content, $parameters ) {
+
+		//Check to see IF a source ($src) is specified
+		if ( ! isset( $parameters[2] ) ) {
+		    return;
+		}
+
+		/*
+		 * Version Check
+		 *
+		 * Check to make sure a Version ($ver) is set
+		 * Otherwise it will Show an error to add a Src (url) to the Enqueued function
+		 */
+		if ( false === isset( $parameters[4] ) || !$parameters[4]['raw'] ) {
+			$this->phpcsFile->addError( 'No Version found for %s; Please supply a value for the fourth argument', $stackPtr, 'MissingVersion', array( $matched_content ) );
+			return;
+		}
+
+		/*
+		 * In footer Check
+		 *
+		 * Check to make sure that $in_footer is set to true
+		 * Otherwise it will warn the user to make sure if its correct
+		 */
+		// print_r($parameters);
+		if ( isset( $parameters[5] ) ) {
```

Comment:
The sniff will now not throw an error if the `$in_footer` param is missing, which I believe it should.

If an explicit `true` or `false` is given, IMHO, the sniff should not report on it at all as that's a conscious developer choice.
If at all, it should be a warning for any non-`true` value, but with a separate errorcode for a `false` value so that particular warning can be silenced easily.